### PR TITLE
Deimos coadd2d debug branchoff

### PR DESCRIPTION
**Pull request by debora-pe**
This PR completes the unfinished work from PR #1235 

- Fixed a bug in coadd2d
- Added the possibility to use `maskdef_offset` as offsets in coadd2d
- Fixed some bugs in manual extraction
- Some documentation

### CODE REVIEW

File: pypeit/core/coadd.py [hunk 1]
```diff
@@ -1987,7 +1990,7 @@ def combspec(waves, fluxes, ivars, masks, sn_smooth_npix,
              const_weights=False, maxiter_reject=5, sn_clip=30.0, lower=3.0, upper=3.0,
              maxrej=None, qafile=None, title='', debug=False,
              debug_scale=False, show_scale=False, show=False,
-             verbose=False):
+             verbose=True):
```
Comment (profxj) [inline pypeit/core/coadd.py:1993]: Just checking that you want to keep this.

**Review by profxj — APPROVED** [review]
Looks good to me!  But I'll admit I haven't recently
touched most parts of what you touched.

**Comment by debora-pe** [comment]
Tests pass.

Dev Suite:
```
--- PYPEIT DEVELOPMENT SUITE PASSED 136/136 TESTS (Masters ignored) ---
Testing Started at 2021-12-04T18:14:31.899897
Testing Completed at 2021-12-05T02:35:21.113792
Total Time: 8:20:49.213895
```

Unit tests:
```
 pytest . -W ignore
============================= test session starts ==============================
platform darwin -- Python 3.8.5, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
rootdir: /Users/dpelliccia/PypeIt/PypeIt, configfile: setup.cfg
collected 304 items                                                            

test_alignments.py .                                                     [  0%]
test_arc.py .                                                            [  0%]
test_archive.py ..                                                       [  1%]
test_arcimage.py ..                                                      [  1%]
test_biasframe.py ..                                                     [  2%]
test_bitmask.py ....                                                     [  3%]
test_bpmimage.py ..                                                      [  4%]
test_bspline.py ...............                                          [  9%]
test_buildimage.py ..                                                    [ 10%]
test_calibrations.py .......                                             [ 12%]
test_chk_calibs.py ..                                                    [ 13%]
test_coadd.py .                                                          [ 13%]
test_collate_1d.py ............                                          [ 17%]
test_cooked.py .                                                         [ 17%]
test_coremeta.py .                                                       [ 18%]
test_datacontainer.py ........                                           [ 20%]
test_detector.py ...                                                     [ 21%]
test_display.py .                                                        [ 22%]
test_edgetrace.py .                                                      [ 22%]
test_fitting.py ....                                                     [ 23%]
test_flatfield.py .                                                      [ 24%]
test_flexure.py ..                                                       [ 24%]
test_flux.py ....                                                        [ 25%]
test_fluxspec.py .....                                                   [ 27%]
test_frametype.py ..                                                     [ 28%]
test_history.py ....                                                     [ 29%]
test_load.py .                                                           [ 29%]
test_load_images.py ................                                     [ 35%]
test_maskdesign_matching.py ...                                          [ 36%]
test_masterframe.py .                                                    [ 36%]
test_match.py ....                                                       [ 37%]
test_metadata.py ......                                                  [ 39%]
test_moment.py ......                                                    [ 41%]
test_msgs.py ..                                                          [ 42%]
test_opticalmodel.py ....                                                [ 43%]
test_parse.py ...                                                        [ 44%]
test_pca.py ........                                                     [ 47%]
test_processimage.py ..                                                  [ 48%]
test_procimg.py .....                                                    [ 49%]
test_pydl.py .                                                           [ 50%]
test_pypeit.py ..                                                        [ 50%]
test_pypeitimage.py ..                                                   [ 51%]
test_pypeitpar.py ................................                       [ 61%]
test_pypeitsetup.py ........                                             [ 64%]
test_qa.py .                                                             [ 64%]
test_runpypeit.py ..                                                     [ 65%]
test_scienceimage.py ...                                                 [ 66%]
test_scripts.py .................                                        [ 72%]
test_setups.py ......................                                    [ 79%]
test_skysub.py ..                                                        [ 79%]
test_slitmask.py ...                                                     [ 80%]
test_slittrace.py ....                                                   [ 82%]
test_spec2dobj.py ......                                                 [ 84%]
test_specobj.py .......                                                  [ 86%]
test_specobjs.py .....                                                   [ 88%]
test_spectrographs.py ....................                               [ 94%]
test_syncspec.py .                                                       [ 95%]
test_telluric.py ..                                                      [ 95%]
test_traceimage.py .                                                     [ 96%]
test_utils.py ...                                                        [ 97%]
test_vcorr.py ..                                                         [ 97%]
test_wavemodel.py .                                                      [ 98%]
test_wavetilts.py ....                                                   [ 99%]
test_wvcalib.py ..                                                       [100%]

======================= 304 passed in 1363.20s (0:22:43) =======================
```

**Review by jhennawi — APPROVED** [review]
Hi @deborape, 

I approve, but you need to clear the conflicts from the merging in of develop. Those conflicts look not so trial. You also need to re-run the dev suite, since I'm a bit concerned that the manual_extract PR (merged into develop previously) and this PR may not play nicely together.

**Comment by debora-pe** [comment]
Tests pass!
Dev Suite failed one test but was due to a crash of python. I re-run that test (without changing anything) and it passed:

```
--- PYPEIT DEVELOPMENT SUITE FAILED 1/136 TESTS (Masters ignored) ---
Failed tests:
    keck_lris_blue_orig/long_600_4000_d500 pypeit (without masters)
Skipped tests:
Testing Started at 2022-01-13T20:20:58.125984
Testing Completed at 2022-01-14T04:43:59.732661
Total Time: 8:23:01.606677
```

```
./pypeit_test develop -i keck_lris_blue_orig
Running tests on the following instruments:
    keck_lris_blue_orig

Reducing data from keck_lris_blue_orig for the following setups:
    long_600_4000_d500

 0 passed/ 0 failed/ 0 skipped STARTED keck_lris_blue_orig/long_600_4000_d500 pypeit (without masters)
 1 passed/ 0 failed/ 0 skipped PASSED  keck_lris_blue_orig/long_600_4000_d500 pypeit (without masters)

--- PYPEIT DEVELOPMENT SUITE PASSED 1/1 TESTS (Masters ignored) ---
Testing Started at 2022-01-14T10:11:56.744639
Testing Completed at 2022-01-14T10:26:57.055244
Total Time: 0:15:00.310605
```

And unit tests:
```
pytest . -W ignore
=============================== test session starts ===============================
platform darwin -- Python 3.8.5, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
rootdir: /Users/dpelliccia/PypeIt/PypeIt, configfile: setup.cfg
collected 307 items                                                               

test_alignments.py .                                                        [  0%]
test_arc.py .                                                               [  0%]
test_archive.py ..                                                          [  1%]
test_arcimage.py ..                                                         [  1%]
test_biasframe.py ..                                                        [  2%]
test_bitmask.py ....                                                        [  3%]
test_bpmimage.py ..                                                         [  4%]
test_bspline.py ...............                                             [  9%]
test_buildimage.py ..                                                       [ 10%]
test_calibrations.py .......                                                [ 12%]
test_chk_calibs.py ..                                                       [ 13%]
test_coadd.py .                                                             [ 13%]
test_collate_1d.py ............                                             [ 17%]
test_cooked.py .                                                            [ 17%]
test_coremeta.py .                                                          [ 17%]
test_datacontainer.py ........                                              [ 20%]
test_detector.py ...                                                        [ 21%]
test_display.py .                                                           [ 21%]
test_edgetrace.py .                                                         [ 22%]
test_fitting.py ....                                                        [ 23%]
test_flatfield.py .                                                         [ 23%]
test_flexure.py ..                                                          [ 24%]
test_flux.py ....                                                           [ 25%]
test_fluxspec.py .....                                                      [ 27%]
test_frametype.py ..                                                        [ 28%]
test_history.py ....                                                        [ 29%]
test_load.py .                                                              [ 29%]
test_load_images.py ................                                        [ 34%]
test_manual.py ..                                                           [ 35%]
test_maskdesign_matching.py ...                                             [ 36%]
test_masterframe.py .                                                       [ 36%]
test_match.py ....                                                          [ 38%]
test_metadata.py ......                                                     [ 40%]
test_moment.py ......                                                       [ 42%]
test_msgs.py ..                                                             [ 42%]
test_opticalmodel.py ....                                                   [ 43%]
test_parse.py ...                                                           [ 44%]
test_pca.py ........                                                        [ 47%]
test_processimage.py ..                                                     [ 48%]
test_procimg.py .....                                                       [ 49%]
test_pydl.py .                                                              [ 50%]
test_pypeit.py ..                                                           [ 50%]
test_pypeitimage.py ..                                                      [ 51%]
test_pypeitpar.py ...............................                           [ 61%]
test_pypeitsetup.py ........                                                [ 64%]
test_qa.py .                                                                [ 64%]
test_runpypeit.py ..                                                        [ 65%]
test_scienceimage.py ...                                                    [ 66%]
test_scripts.py ..................                                          [ 71%]
test_setups.py ......................                                       [ 79%]
test_skysub.py ..                                                           [ 79%]
test_slitmask.py ...                                                        [ 80%]
test_slittrace.py ....                                                      [ 82%]
test_spec2dobj.py ......                                                    [ 84%]
test_specobj.py .......                                                     [ 86%]
test_specobjs.py .....                                                      [ 87%]
test_spectrographs.py ....................                                  [ 94%]
test_syncspec.py .                                                          [ 94%]
test_telluric.py ..                                                         [ 95%]
test_traceimage.py .                                                        [ 95%]
test_utils.py ....                                                          [ 97%]
test_vcorr.py ..                                                            [ 97%]
test_wavemodel.py .                                                         [ 98%]
test_wavetilts.py ....                                                      [ 99%]
test_wvcalib.py ..                                                          [100%]

======================== 307 passed in 2197.26s (0:36:37) =========================
```

**Comment by debora-pe** [comment]
merging

**Pull request merged**